### PR TITLE
Support ISO 8601 dates, use hyphens as separators

### LINKED ISSUE #1: Cannot handle ISO 8601 dates
I used the great bookmarklet on https://musicbrainz.org/release/5b39bd29-c7aa-40e9-93d0-1b1b60a71140/edit and it replaced

    1987<HYPHEN-MINUS>07<HYPHEN-MINUS>30

with 

    1987<EN DASH>07<HYPHEN>30

Same if I paste this date in any track title.

### PR DESCRIPTION
**Pull request by kellnerd**
Closes #1

### CODE REVIEW

File: src/guessUnicodePunctuation.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ function guessUnicodePunctuation() {
 		[/'/g, '’'], // ... and finally the apostrophes should be remaining
 		[/(?<!\.)\.{3}(?!\.)/g, '…'], // horizontal ellipsis (but not more than three dots)
 		[/ - /g, ' – '], // en dash as separator
+		[/(\d{4})-(\d{2})-(\d{2})/g, '$1‐$2‐$3'], // hyphens for ISO 8601 dates, e.g. 1987‐07–30
+		[/(\d{4})-(\d{2})/g, '$1‐$2'], // hyphen for ISO 8601 partial dates, e.g. 2016-04
 		[/(\d+)-(\d+)/g, '$1–$2'], // en dash for ranges where it means "to", e.g. 1965–1972
```
Comment (kellnerd) [inline src/guessUnicodePunctuation.js:44]: This should work for all full or partial ISO 8601 dates but I would have preferred a more compact solution that only requires one additional regex search:
```suggestion
		[/(\d{4})-(\d{2})/g, '$1‐$2'], // hyphen for ISO 8601 dates (YYYY-MM-DD), e.g. 2016-04
		[/(?<!‐)(\d+)-(\d+)/g, '$1–$2'], // en dash for ranges, e.g. 1965–1972 (but not the MM‐DD part of dates)
```
Unfortunately this specific negative lookbehind for a hyphen does not work, it ignores the leading hyphen before MM-DD (that was replaced by the previous regex) nevertheless and treats it as a range 😕

@jesus2099 Do you have an idea why this does not work as expected?

**Comment by jesus2099** [comment]
These date detections can be very tricky when you want to have date ranges in mind too.

I can only show you what I have done in the past, but not sure if it will help you:

First I declared some shortcuts for date elements:

https://github.com/jesus2099/konami-command/blob/82513d11024abbab40237282e10d5ed9ee5d018e/mb_SUPER-MIND-CONTROL-II-X-TURBO.user.js#L48-L56

Then here is a place around where I want to detect a date (complete or not):

https://github.com/jesus2099/konami-command/blob/82513d11024abbab40237282e10d5ed9ee5d018e/mb_SUPER-MIND-CONTROL-II-X-TURBO.user.js#L542

Then here is a place around where I want to detect either a date or a date range (both complete or not, IIRC), but in a context (prompt) where a strict date format is requested to the user to type:

https://github.com/jesus2099/konami-command/blob/82513d11024abbab40237282e10d5ed9ee5d018e/mb_SUPER-MIND-CONTROL-II-X-TURBO.user.js#L1228

**Comment by jesus2099** [comment]
There is also an old excellent [Paste-A-Date!](https://bitbucket.org/96187/userscripts/src/master/paste-a-date/) userscript by [nikki](https://musicbrainz.org/user/nikki), also known as 96187, nstars and [stars](https://web.archive.org/web/20131103215822/userscripts.org/users/41307).
This script can detect text dates. I know it's irrelevant to your need but maybe it contains useful hints as she was the very clever one thanks to whom I discovered userscripts.

**Review by kellnerd — COMMENTED** [review]
Thank you for the links, the _Paste-A-Date!_ script indeed has some clever tricks... date detection is really a complex topic if you want to catch all edge cases.
As it is only one rather rare substitution for the title punctuation guess script, I would like to keep the code rather minimal, because there will always be cases which have to be corrected manually.

**Pull request merged**